app/src/main/java/woowacourse/shopping/view/order/vm/OrderViewModelFactory.kt

```
@@ -0,0 +1,28 @@
+    private val couponApplierFactory: CouponApplierFactory,
+) : ViewModelProvider.Factory {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        if (modelClass.isAssignableFrom(OrderViewModel::class.java)) {
+            return OrderViewModel(
+                couponRepository,
+                orderRepository,
+                couponValidator,
+                couponApplierFactory,
+            ) as T
+        } else {
+            throw IllegalArgumentException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)